ui/src/app/pages/access/data-access-requirements.tsx

```diff
@@ -716,7 +711,16 @@ export const DataAccessRequirements = fp.flow(withProfileErrorModal)(
       );
 
       syncModulesPromise
-        .then(async () => await reload())
+        .then(async (syncedModules) => {
+          // Bug 26 Feb 2025: reloading can cause an infinite loop in *some* cases,
+          // so we only reload when needed
+
+          // we don't want to sync for no-op cases which look like []
+          // we also don't want to sync for error cases, which we know can look like [undefined]
+          if (syncedModules?.filter((x) => x !== undefined).length !== 0) {
```

Comment:
do we want to log some error if sync fails?

Reply:
these errors are in the API calls, so we will see them in the logs for that reason.  Example: https://console.cloud.google.com/logs/query;query=SEARCH%2528%22syncComplianceTrainingStatus%22%2529%0A-protoPayload.serviceName%3D%22logging.googleapis.com%22;cursorTimestamp=2025-02-26T20:49:26.442392Z;duration=PT1H?invt=Abqokw&project=all-of-us-workbench-test&inv=1